src/js/components/devices/terminal.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from '@material-ui/core';
+
+import { XTerm } from 'xterm-for-react';
+import { FitAddon } from 'xterm-addon-fit';
+import { SearchAddon } from 'xterm-addon-search';
+import { encode, decode } from 'msgpack5';
+
+export const Terminal = props => {
+  const { deviceId } = props;
+  const [socket, setSocket] = useState(null);
+  const xtermRef = React.useRef(null);
+
+  const onData = data => {
+    const msg = { cmd: 'shell', data: data };
+    const encodedData = encode(msg);
+    socket.send(encodedData);
```

Comment:
for some crazy reasons I am getting here:
```js
terminal.js:19 Uncaught TypeError: (0 , msgpack5__WEBPACK_IMPORTED_MODULE_5__.encode) is not a function
    at Object.onData (terminal.js:19)
    at t.onData (bundle.js:14)
    at e.fire (bundle.js:1)
    at bundle.js:1
    at e.fire (bundle.js:1)
    at e.triggerDataEvent (bundle.js:1)
    at t._keyDown (bundle.js:1)
    at HTMLTextAreaElement.<anonymous> (bundle.js:1)
```
which was not the case with 
```js
import { encode, decode } from '@msgpack/msgpack';
```
and 
```js
"@msgpack/msgpack": "^1.12.2",
```

Reply:
I switched to msgpack5, which is a better maintained implementation of the messagepack protocol.
By the way, it's the one which is suggested on the msgpack website.